### PR TITLE
[ROCm] Reverting ROCm to use MIOpen Find Mode APIs (be default) for convolution

### PR DESCRIPTION
**Pull request by deven-amd**
This PR reverts ROCm to use MIOpen Find Mode APIs (be default) for convolution.  The use of MIOpen Immediate Mode API (instead of the Find Mode APIs) can be specified by the setting the env var `TF_ROCM_USE_IMMEDIATE_MODE=1`.

Almost all of the changes in this PR are within code that is specific to the ROCm platform, so this PR should not have any impact on non ROCm builds.

----------------

/cc @chsigg @whchung

### CODE REVIEW

**Comment by gbaned** [comment]
@deven-amd Can you please resolve conflicts? Thanks!

**Comment by deven-amd** [comment]
@gbaned, I have rebased the PR to resolve the merge conflicts

**Comment by deven-amd** [comment]
@gbaned, gentle ping

**Comment by deven-amd** [comment]
gentle ping

**Comment by deven-amd** [comment]
gentle ping

File: tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc [hunk 1]
```diff
@@ -118,11 +118,11 @@ std::vector<AlgorithmDesc> GetAlgorithms(CudnnConvKind kind,
   return algorithms;
 }
 
-StatusOr<std::vector<se::dnn::ProfileResult>> GetAlgorithms(
+StatusOr<std::vector<se::dnn::ProfileResult>> GetMIOpenAlgorithms(
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc]: I'm not sure this change makes sense, this function is used for both ROCm and CUDA? Isn't MIOpen ROCm-specific?

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc (hunk 1 above)
Comment (deven-amd) [reply tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc]: this version of the `GetAlgorithms` function is used on the ROCM side only, which is why it is being renamed to `GetMIOpenAlgorithms` (to make the ROCm specificness explicit)

Comment (cheshire) [reply tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc]: Makes sense!

**Review by cheshire — DISMISSED** [review]

**Comment by gbaned** [comment]
@deven-amd Can you please resolve conflicts? Thanks!

**Comment by deven-amd** [comment]
@gbaned rebase done to remove merge conflicts.

@cheshire please re-approve.

thanks again

**Review by cheshire — APPROVED** [review]

**Comment by gbaned** [comment]
@deven-amd Still, conflicts appearing, can you please resolve those? Thanks!

**Comment by deven-amd** [comment]
@gbaned , it seems like this PR did get merged (but not marked as such) and is now causing merge conflicts with itself. Assuming this is correct, how do we go about marking this PR as "merged".

thanks

**Comment by chsigg** [comment]
It was merged in c6667ea, but I'm not aware of any way to mark this PR as merged manually. I think the only option is to just close it.

**Pull request closed without merging**